File: 1813-maximum-erasure-value/1813-maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        int i=0;
        int j=0;
        int res=0;
        int sum=0;
        unordered_set<int> st;

        while(j<nums.size()){
            if(!st.count(nums[j])){
                sum+=nums[j];
                res=max(res,sum);
                st.insert(nums[j]);
                j++;
            }
            else{
                while(st.count(nums[j])){
                    sum-=nums[i];
                    st.erase(nums[i]);
                    i++;
                }
            }
        }
        return res;
    }
};

File: 1813-maximum-erasure-value/1813-maximum-erasure-value.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        Set<Integer> st = new HashSet<>();
        int i=0;
        int j=0;
        int res=0;
        int sum=0;

        while(j<nums.length){
            if(!st.contains(nums[j])){
                sum+=nums[j];
                st.add(nums[j]);
                res=Math.max(res,sum);
                j++;
            }
            else{
                while (st.contains(nums[j])){
                    sum-=nums[i];
                    st.remove(nums[i]);
                    i++;
                }
            }
        }
        return res;
    }
}